### PR TITLE
Add binding collection/map to SdkBindingDataFactory for Scala

### PR DESCRIPTION
**Pull request by narape**
# TL;DR
Add methods to `SdkBindingDataFactory` to create `SdkBindingData` from collection or maps of other `SdkBindingData`

## Type
 - [ ] Bug Fix
 - [X] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [X] Unit tests added
 - [X] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
We were missing a way to construct `SdkBindingData` from collections (or maps) of other `SdkBindingData`´s in the scalaś `SdkBindingDataFactory`, as we have in the java one.

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

**Review by andresgomezfrr — APPROVED** [review]
🚀 LGTM
